platform/node/src/node_map.cpp

```diff
@@ -579,6 +582,68 @@ void NodeMap::RemoveLayer(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 
     nodeMap->map->removeLayer(*Nan::Utf8String(info[0]));
 }
+    
+void NodeMap::AddImage(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    using namespace mbgl::style;
+    using namespace mbgl::style::conversion;
+    
+    auto nodeMap = Nan::ObjectWrap::Unwrap<NodeMap>(info.Holder());
+    if (!nodeMap->map) return Nan::ThrowError(releasedMessage());
+    
+    if (info.Length() != 2) {
+        return Nan::ThrowTypeError("Two arguments required");
+    }
+    
+    if (!info[0]->IsString()) {
+        return Nan::ThrowTypeError("First argument must be a string");
+    }
+
+    if (!info[1]->IsObject()) {
+        return Nan::ThrowTypeError("Second argument must be an object");
+    }
+    
+    auto imageObject = Nan::To<v8::Object>(info[1]).ToLocalChecked();
+    
+    if (!Nan::Get(imageObject, Nan::New("height").ToLocalChecked()).ToLocalChecked()->IsNumber()) {
+        return Nan::ThrowTypeError("height parameter required");
+    }
+    
+    if (!Nan::Get(imageObject, Nan::New("width").ToLocalChecked()).ToLocalChecked()->IsNumber()) {
+        return Nan::ThrowTypeError("width parameter required");
+    }
+    
+    if (!Nan::Get(imageObject, Nan::New("imageBuffer").ToLocalChecked()).ToLocalChecked()->IsObject()) {
+        return Nan::ThrowTypeError("imageBuffer parameter required");
+    }
+    
+    double imageHeight = Nan::Get(imageObject, Nan::New("height").ToLocalChecked()).ToLocalChecked()->NumberValue();
```

Comment:
`imageHeight` and `imageWidth` should be integers -- validate the type using `IsUint32` rather than `IsNumber` and convert using `Uint32Value` rather than `NumberValue`, assigning to a `uint32_t` rather than a `double`.